server/src/test/java/org/elasticsearch/cluster/coordination/ClusterFormationFailureHelperTests.java

```diff
@@ -71,7 +71,7 @@ public void testScheduling() {
                 warningCount.incrementAndGet();
                 return new ClusterFormationState(Settings.EMPTY, clusterState, emptyList(), emptyList(), 0L);
             },
-            deterministicTaskQueue.getThreadPool());
```

Comment:
I think it'd be good to test that the runnable we pass in is actually called when we think it should be.

Reply:
Sure, https://github.com/elastic/elasticsearch/pull/39756/commits/452af96a1c254f34525b03621c192bb6247bae3f